pkg/resource/deploy/step_executor.go

```diff
@@ -366,13 +367,15 @@ func (se *stepExecutor) cancelDueToError(err error, step Step) {
 	if !set {
 		logging.V(10).Infof("StepExecutor already recorded an error then saw: %v", err)
 	}
-	if se.opts.ContinueOnError {
+	if se.ignoreErrors {
+		// Do nothing.
+	} else if se.deployment.opts.ContinueOnError {
 		step.Fail()
 		// Record the failure, but allow the deployment to continue.
 		se.erroredStepLock.Lock()
 		defer se.erroredStepLock.Unlock()
 		se.erroredSteps = append(se.erroredSteps, step)
-	} else if !se.ignoreErrors {
+	} else {
```

Comment:
This logic has been reordered to avoid the previous mutation where we set `ContinueOnError` to false for the duration of the step (incidentally, that no longer works without also reverting `ContinueOnError` to its previous value since we now (correctly?) share options between all the pieces of the process without copying (though again, copying is hardly a bug -- mutation is/was the problem)